pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml

```diff
@@ -102,7 +102,10 @@ angular: teacherDashboard
       //  'remove': {method:'DELETE'},
       //  'delete': {method:'DELETE'}
          update: {method:'POST', url: 'v2/sections/:id/update'},
+         // TODO: This isn't very restful
+         allStudents: {method:'GET', url:'v2/sections/:id/students', isArray: true},
          addStudents: {method:'POST', url:'/v2/sections/:id/students', isArray: true},
+         moveStudents: {method:'POST', url:'/sections/:new_section_code/transfers'},
```

Comment:
The multiple layers of templating makes this a bit annoying to think about
-- but one thing that helps is that haml happens first and then angular. So
I think you can generate the url with studio_url in haml with the string
":new_section_code", not the section code in it, and then angular will fill
that in.

On Sat, Jan 9, 2016 at 11:24 PM, David Wu <notifications@github.com> wrote:

> In pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml
> <https://github.com/code-dot-org/code-dot-org/pull/3662#discussion_r49272616>
> :
>
> >           addStudents: {method:'POST', url:'/v2/sections/:id/students', isArray: true},
> > +         moveStudents: {method:'POST', url:'/sections/:new_section_code/transfers'},
>
> Ah, so the new section code isn't figured out until the user enters it, so
> I cannot pre-generate the URL in haml. I'll probably have to change the API
> endpoint to take in the new section code as a param so I can use the
> CDO.studio_url method. I'll fix that up.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/3662/files#r49272616>.
>